src/rest/SearchEndpoint.ts

```diff
@@ -1073,26 +1081,53 @@ export class SearchEndpoint implements ISearchEndpoint {
       }
     });
 
+    const request = () => this.caller.call(params);
+
     try {
-      const response = await this.caller.call(params);
+      const response = await request();
       return response.data as T;
     } catch (error) {
-      if (autoRenewToken && this.accessToken.isExpired(error)) {
-        const renewSuccess = await this.accessToken.doRenew();
-        if (renewSuccess) {
-          return this.performOneCall(params, callOptions, autoRenewToken) as Promise<T>;
-        }
-      } else if (error.statusCode == 0 && this.isRedirecting) {
-        // The page is getting redirected
-        // Set timeout on return with empty string, since it does not really matter
-        _.defer(function() {
-          return '';
-        });
+      const tokenWasRenewed = await this.renewAccessTokenIfExpired(error);
```

Comment:
I'd expect the code to read more like a big switch case on the error code, versus trying one error handler function. and if it does not work, go to the next error handler.

Seems much more straightforward to execute all error handler only when they are actually needed.